Marlin/src/gcode/gcode.cpp

```diff
@@ -701,8 +701,7 @@ void GcodeSuite::process_next_command() {
    */
   void GcodeSuite::process_subcommands_now_P(const char *pgcode) {
     // Save the parser state
-    char saved_cmd[strlen(parser.command_ptr) + 1];
-    strcpy(saved_cmd, parser.command_ptr);
+    char *saved_cmd = parser.command_ptr;
```

Comment:
This isn't going to work because the contents of `parser.command_ptr` are trounced by any later call to `parser.parse()`. And, `parser.command_ptr` always points to the same buffer, just slightly different offsets within it. This function very much needs to save the entire command string, and if this function is meant to be invoked recursively, then it has to be on the stack, or else it has to create a large global static FILO buffer to hold all the saved strings.